File: microservice/artistquery/src/main/java/vn/io/echovibe/artist/query/dto/ArtistDto.java
package vn.io.echovibe.artist.query.dto;

import java.time.Instant;
import java.util.ArrayList;
import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;
import lombok.experimental.SuperBuilder;
import vn.io.echovibe.core.dto.QueryDto;

@Getter
@Setter
@ToString
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder
public class ArtistDto extends QueryDto {
  private String id;

  private String urn;

  private String name;

  private String market;

  private String biography;

  private String description;

  @Builder.Default private Boolean isPublic = true;

  @Builder.Default private Boolean isPublished = false;

  private String thumbnailUrl;

  private String backgroundUrl;

  @Builder.Default private List<String> tags = new ArrayList<>();

  private String refCode;

  private String createdBy;

  private String updatedBy;

  private Instant createdAt;

  private Instant updatedAt;
}

File: ui/adminweb/src/app/dto/artist-dto.ts
import { ArtistImage, ArtistRevision } from '../model/artist';
import { Tag } from '../model/tag';

export interface ArtistProfileDto {
  name: string | null;
  description: string | null;
  biography: string | null;
  thumbnailUrl: string | null;
  backgroundUrl: string | null;
  nationalityIsoCode: string | null;
}

export interface CreateArtistDto {
  profile: ArtistProfileDto;
  refCode: string | null;
  isPublic: boolean;
  tags: Tag[];
}

export interface DeleteArtistDto {
  id: string;
}

export interface ReleaseArtistDto {
  id: string;
}

export interface ArtistDto {
  id: string;
  urn: string;
  refCode: string | null;
  profile: ArtistProfileDto;
  isPublic: boolean;
  isReleased: boolean;
  isVerified: boolean;
  images: ArtistImage[] | null;
  revisionNumber: number;
  revisions: ArtistRevision[] | null;
  tags: Tag[];
  createdAt: string;
  updatedAt: string;
  createdBy: string | null;
  updatedBy: string | null;
}

export interface UpdateArtistDto {
  id: string;
  refCode: string | null;
  isPublic: boolean;
  tags: Tag[];
  profile: ArtistProfileDto;
}
